src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -1395,9 +803,42 @@ protected virtual void EmitMarshalElementNativeToManaged()
             LoadManagedValue(codeStream);
         }
 
+        protected virtual void EmitMarshalFieldManagedToNative()
+        {
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+            ILCodeStream marshallingCodeStream = _ilCodeStreams.MarshallingCodeStream;
+
+            SetupArgumentsForFieldMarshalling();
+
+            StoreManagedValue(marshallingCodeStream);
```

Comment:
This is also quite unfortunate. We should brainstorm to see if we can avoid the copy. But that can happen later.